docs/DeveloperGuide.md

```diff
@@ -188,6 +198,29 @@ From the diagram illustrated above:
 1. `LogicManager` would subsequently invoke the `execute()` method of the `UnpoolCommand` which in turn invokes `deletePool()` method with an argument of `1`.
 1. This would update the model by deleting the specified pool at the first index, then the result of the command execution `CommandResult` would be created and returned back to `LogicManager`.
 
+### findPool feature
+This feature allows users to find a pool that contains a passenger with a provided keyword in their name. 
+
+Given below is the Sequence Diagram for interactions within the Logic component for the `execute("findPool n/Alice")` command.
+![Interactions Inside the Logic Component for the `findPool n/Alice` Command](images/FindPoolSequenceDiagram.png)
+
+<div markdown="span" class="alert alert-info">:information_source: **Note:**  The `command` argument that is passed into
+`execute()`, represents the string `"findPool n/Alice"`, and has been abstracted for readability.
+<br>
+The lifeline for `FindPoolCommandParser` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
+</div>
+
+From the diagram illustrated above:
+1. `LogicManager` has its `execute()` method called when a user enters the `"findPool n/Alice"` command.
+1. `AddressBookParser` class is then created, which subsequently creates `FindPoolCommandParser` class to help parse the user's command.
+1. `AddressBookParser` would then have its `parse()` method invoked to parse the argument `"n/Alice"` and passes it to
+   `FindPoolCommandParser`.
+1. `FindPoolCommandParser` parses the argument `"n/Alice"` and creates a PooledPassengerContainsKeywordPredicate which is returned to the FindPoolCommandParser.
+1. `FindPoolCommandParser` then creates a `FindPoolCommand`, and provides the newly created `PooledPassengerContainsKeywordPredicate` to it. The FindPoolCommand object is then returned to LogicManager.
+1. `LogicManager` would subsequently invoke the `execute()` method of the `FindPoolCommand`, which turn calls the `updateFilteredPoolList()` method in `Model`, causing the shown pool list to be updated according to the predicate in the FindPoolCommand object.
```

Comment:
The word `FindPoolCommand` at the end of the line should be formatted as code block?